whylog/assistant/regex_assistant/regex.py

```diff
@@ -0,0 +1,116 @@
+"""
+Regex verification and creating (but not finding groups in regex)
+"""
+import re
+
+special_characters = frozenset(['.', '^', '$', '*', '+', '{', '}', '?', '[', ']', '|', '(', ')'])
+
+
+class NoDateGroupError(Exception):
+    def __init__(self, line_content, regex, line_index):
+        self.line_content = line_content
+        self.regex = regex
+        self.line_index = line_index
+
+    def __str__(self):
+        return 'No date group in regex. Regex: %s. Line content: %s' % (
+            self.regex, self.line_content
+        )
+
+
+class NotMatchingRegexError(Exception):
+    def __init__(self, line_content, regex):
+        self.line_content = line_content
+        self.regex = regex
+
+    def __str__(self):
+        return 'Regex do not match line. Regex: %s. Line content: %s' % (
+            self.regex, self.line_content
+        )
+
+
+class DateFromFutureError(Exception):
+    def __init__(self, parsed_date, date_text):
+        self.date_text = date_text
+        self.parsed_date = parsed_date
+
+    def __str__(self):
+        return 'Date is from future. date text %s,  parsed date %s' % (
```

Comment:
11 lines later an exception shows field in form of `Field: %s. Field: %s.` and here it is `field %s,  field %s', with a double space. Please make it consistent.